klippy/extras/virtual_sdcard.py

```diff
@@ -160,7 +160,7 @@ def cmd_M23(self, gcmd):
         self._reset_file()
         try:
             orig = gcmd.get_commandline()
-            filename = orig[orig.find("M23") + 4:].split()[0].strip()
+            filename = orig.split(None, 1)[1].strip()
```

Comment:
If the original command contains a line number then I suspect that this will produce an invalid filename.  I suspect that what you desire could be accomplished by adding `maxsplit=1` to the original line, ie:
```python
filename = orig[orig.find("M23") + 4:].split(maxsplit=1)[0].strip()
```

Reply:
Thanks, I was unaware that gcode line numbers were a thing or that Klipper supported them.

That stated, one thing I forgot to mention was that the current code breaks if the `M23` is renamed to anything other than `M23\d` (where `\d` is a single digit). I didn't realize that until after I posted the PR, but I assume it's a bug worth fixing as well, so I would have to tweak your suggestion a bit to just skip a line number (if present).